docs/reference-config.yaml

```diff
@@ -50,6 +50,7 @@ kafka:
     mechanism: "PLAIN"
     # GSSAPI / Kerberos config properties
     gssapi:
+      # Required. One of USER_AUTH or KEYTAB_AUTH
```

Comment:
I'm not sure which of the below are actually required with either of the authTypes, or where the empty defaults are ok, but it would be nice to say in the doc.